### PR TITLE
etcdserver: better v3 api error handling

### PR DESCRIPTION
**Pull request by xiang90**
Start to handle error

### CODE REVIEW

File: etcdserver/v3demo_server.go [hunk 1]
```diff
@@ -127,29 +138,41 @@ func applyTxn(kv dstorage.KV, rt *pb.TxnRequest) *pb.TxnResponse {
...
+	return resp, err
 }
 
 func applyUnion(kv dstorage.KV, union *pb.RequestUnion) *pb.ResponseUnion {
 	switch {
 	case union.RequestRange != nil:
-		return &pb.ResponseUnion{ResponseRange: applyRange(kv, union.RequestRange)}
+		resp, err := applyRange(kv, union.RequestRange)
+		if err != nil {
+			panic("unexpected error during txn")
```
Comment (yichengq) [inline etcdserver/v3demo_server.go]: why does it panic here? It may be ErrCompacted, right?

Comment (xiang90) [inline etcdserver/v3demo_server.go]: we should check potential errors before executing the transaction. we cannot abort executed commands. 

see line 120 for more details.

Comment (yichengq) [inline etcdserver/v3demo_server.go]: I see. So the point is that as long as txn starts, it should be successful.

**Comment by yichengq** [comment]
LGTM

File: etcdserver/v3demo_server.go (hunk 1 above)
Comment (xiang90) [inline etcdserver/v3demo_server.go]: It MUST be.

**Pull request merged**